dotnet/src/Microsoft.Agents.AI.Abstractions/Features/IAgentFeatureCollection.cs

```diff
@@ -24,23 +25,30 @@ public interface IAgentFeatureCollection : IEnumerable<KeyValuePair<Type, object
     int Revision { get; }
 
     /// <summary>
-    /// Gets or sets a given feature. Setting a null value removes the feature.
+    /// Attempts to retrieve a feature of the specified type.
     /// </summary>
-    /// <param name="key"></param>
-    /// <returns>The requested feature, or null if it is not present.</returns>
-    object? this[Type key] { get; set; }
+    /// <typeparam name="TFeature">The type of the feature to retrieve.</typeparam>
+    /// <param name="feature">When this method returns, contains the feature of type <typeparamref name="TFeature"/> if found; otherwise, the
+    /// default value for the type.</param>
+    /// <returns>
+    /// <see langword="true"/> if the feature of type <typeparamref name="TFeature"/> was successfully retrieved;
+    /// otherwise, <see langword="false"/>.
+    /// </returns>
+    bool TryGet<TFeature>([MaybeNullWhen(false)] out TFeature feature)
+        where TFeature : class;
```

Comment:
```suggestion
    bool TryGet<TFeature>([MaybeNullWhen(false)] out TFeature? feature)
        where TFeature : notnull;
```

Would it be better to enforce `notnull` here and explicitly add `?` in the parameter list?